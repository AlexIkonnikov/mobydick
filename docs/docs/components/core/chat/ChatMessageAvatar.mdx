import ChatMessage from './assets/ChatMessage.png';

<img src={ChatMessage} />

Reusable ChatMessage component

## Example

```tsx
import { ChatMessageAvatar } from '@lad-tech/mobydick-core';

export const ChatWidget = () => {
  return (
    <ChatMessageAvatar
      user={{
        firstName: 'firstName',
        lastName: 'lastName',
        middleName: 'middleName',
      }}
      isMe={false}
      time={'12-12-1998'}
      message={'message'}
    />
  );
};

```

## `Props`

### [`IAvatar Props`](../others/Avatar#props)

Inherits [IAvatar Props](../others/Avatar#props).

### [`IChatMessage Props`](./ChatMessage#props)

Inherits [IChatMessage Props](./ChatMessage#props).

### `onPress`

| TYPE       |
|:-----------|
| () => void |
