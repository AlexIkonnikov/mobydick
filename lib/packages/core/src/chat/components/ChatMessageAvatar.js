"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Avatar_1 = __importDefault(require("../../other/components/Avatar/Avatar"));
const TouchableOpacity_1 = __importDefault(require("../../basic/components/TouchableOpacity/TouchableOpacity"));
const other_1 = require("../../other");
const View_1 = __importDefault(require("../../basic/components/View/View"));
const useStyles_1 = __importDefault(require("../../styles/hooks/useStyles"));
const styles_1 = require("../../styles");
const ChatMessage_1 = __importDefault(require("./ChatMessage"));
const ChatMessageAvatar = (props) => {
    const { onPress, isMe, message, image, time, ...otherProps } = props;
    const [styles] = (0, useStyles_1.default)(stylesCreate, isMe);
    return (<View_1.default style={styles.container}>
      <TouchableOpacity_1.default onPress={onPress} style={styles.avatar} disabled={onPress === undefined}>
        <Avatar_1.default size={other_1.IAvatarSize.S} {...otherProps}/>
      </TouchableOpacity_1.default>
      <ChatMessage_1.default isMe={isMe} message={message} time={time} image={image}/>
    </View_1.default>);
};
const stylesCreate = (0, styles_1.createStyles)(({ spaces }, isMe) => ({
    container: {
        flexDirection: 'row',
        alignSelf: isMe ? 'flex-end' : 'flex-start',
    },
    avatar: {
        paddingRight: spaces.Space8,
        marginVertical: spaces.Space4,
        alignSelf: 'flex-end',
    },
}));
exports.default = ChatMessageAvatar;
