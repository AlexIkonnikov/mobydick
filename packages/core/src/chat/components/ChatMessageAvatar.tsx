import {StyleSheet} from 'react-native';

import Avatar from '../../other/components/Avatar/Avatar';
import TouchableOpacity from '../../basic/components/TouchableOpacity/TouchableOpacity';
import {IAvatarSize} from '../../other';
import {IChatMessageAvatar} from '../types';
import View from '../../basic/components/View/View';
import useStyles from '../../styles/hooks/useStyles';
import {IThemeContext} from '../../styles';

import ChatMessage from './ChatMessage';

const ChatMessageAvatar = (props: IChatMessageAvatar) => {
  const {onPress, isMe, message, image, time, ...otherProps} = props;
  const [styles] = useStyles(stylesCreate, isMe);

  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={onPress}
        style={styles.avatar}
        disabled={onPress === undefined}>
        <Avatar size={IAvatarSize.S} {...otherProps} />
      </TouchableOpacity>
      <ChatMessage isMe={isMe} message={message} time={time} image={image} />
    </View>
  );
};

const stylesCreate = ({spaces}: IThemeContext, isMe: boolean) =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignSelf: isMe ? 'flex-end' : 'flex-start',
    },
    avatar: {
      paddingRight: spaces.Space8,
      marginVertical: spaces.Space4,
      alignSelf: 'flex-end',
    },
  });

export default ChatMessageAvatar;
