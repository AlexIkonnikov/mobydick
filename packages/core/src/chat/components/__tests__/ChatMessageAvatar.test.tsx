import {render} from '@testing-library/react-native';
import React from 'react';

import ChatMessageAvatar from '../ChatMessageAvatar';
const userWithoutPhoto = {
  firstName: 'Иван',
  lastName: 'Пушкин',
};

describe('@npm/mobydick-core/ChatMessageAvatar', () => {
  it('renders correctly isNotMe', () => {
    const {toJSON} = render(
      <ChatMessageAvatar
        user={userWithoutPhoto}
        message={'message'}
        isMe={false}
        time={'12:12'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly onPress', () => {
    const {toJSON} = render(
      <ChatMessageAvatar
        onPress={() => null}
        user={userWithoutPhoto}
        message={'message'}
        isMe={true}
        time={'12:12'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
