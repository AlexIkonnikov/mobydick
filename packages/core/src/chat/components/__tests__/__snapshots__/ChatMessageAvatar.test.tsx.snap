// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/ChatMessageAvatar renders correctly isNotMe 1`] = `
<View
  style={
    {
      "alignSelf": "flex-start",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "flex-end",
        "marginVertical": 6,
        "opacity": 1,
        "paddingRight": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "borderRadius": 18,
            "height": 36,
            "justifyContent": "center",
            "width": 36,
          },
          {
            "backgroundColor": "#2B78EE",
          },
          undefined,
          {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        allowFontScaling={true}
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Inter-Regular",
              "fontSize": 15,
              "lineHeight": 21,
            },
            undefined,
          ]
        }
      >
        ИП
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignSelf": "flex-start",
        "flexDirection": "row",
        "maxWidth": "90%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFF",
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 15,
          "borderRadius": 15,
          "marginVertical": 6,
        }
      }
    >
      <View
        style={
          {
            "paddingBottom": 6,
            "paddingHorizontal": 15,
            "paddingTop": 12,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#444B5A",
                "fontFamily": "Inter-Regular",
                "fontSize": 21,
                "lineHeight": 27,
              },
              undefined,
            ]
          }
        >
          message
        </Text>
      </View>
      <View
        style={
          {
            "alignSelf": "flex-end",
            "paddingBottom": 12,
            "paddingHorizontal": 9,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#9BA1B0",
                "fontFamily": "Inter-Regular",
                "fontSize": 15,
                "lineHeight": 21,
              },
              undefined,
            ]
          }
        >
          12:12
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`@lad-tech/mobydick-core/ChatMessageAvatar renders correctly onPress 1`] = `
<View
  style={
    {
      "alignSelf": "flex-end",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "flex-end",
        "marginVertical": 6,
        "opacity": 1,
        "paddingRight": 12,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "borderRadius": 18,
            "height": 36,
            "justifyContent": "center",
            "width": 36,
          },
          {
            "backgroundColor": "#2B78EE",
          },
          undefined,
          {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        allowFontScaling={true}
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Inter-Regular",
              "fontSize": 15,
              "lineHeight": 21,
            },
            undefined,
          ]
        }
      >
        ИП
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignSelf": "flex-end",
        "flexDirection": "row",
        "maxWidth": "90%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#2B78EE",
          "borderBottomLeftRadius": 15,
          "borderBottomRightRadius": 0,
          "borderRadius": 15,
          "marginVertical": 6,
        }
      }
    >
      <View
        style={
          {
            "paddingBottom": 6,
            "paddingHorizontal": 15,
            "paddingTop": 12,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Inter-Regular",
                "fontSize": 21,
                "lineHeight": 27,
              },
              undefined,
            ]
          }
        >
          message
        </Text>
      </View>
      <View
        style={
          {
            "alignSelf": "flex-end",
            "paddingBottom": 12,
            "paddingHorizontal": 9,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "rgba(255, 255, 255, 0.75)",
                "fontFamily": "Inter-Regular",
                "fontSize": 15,
                "lineHeight": 21,
              },
              undefined,
            ]
          }
        >
          12:12
        </Text>
      </View>
    </View>
  </View>
</View>
`;
